src/Lykke.Service.CandlesHistory.Services/Candles/CandlesChecker.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Common.Log;
+using Lykke.Service.CandlesHistory.Core.Domain.Candles;
+using Lykke.Service.CandlesHistory.Core.Services;
+using Lykke.Service.CandlesHistory.Core.Services.Candles;
+using Lykke.Service.CandlesHistory.Services.Settings;
+using System;
+using System.Collections.Generic;
+
+namespace Lykke.Service.CandlesHistory.Services.Candles
+{
+    public class CandlesChecker : ICandlesChecker
+    {
+        private readonly ILog _log;
+        private readonly IClock _clock;
+        private readonly ICandlesHistoryRepository _candlesHistoryRepository;
+        private readonly ErrorManagementSettings _errorSettings;
+
+        private Dictionary<string, DateTime> _knownUnsupportedAssetPairs;
+
+        public CandlesChecker(
+            ILog log,
+            IClock clock,
+            ICandlesHistoryRepository _historyRep,
+            ErrorManagementSettings settings)
```

Comment:
Do not inject settings directly to the service. Consider to inject timeout value independently. Insted of using ```NotifyOnCantStoreAssetPair``` here, you can introduce separate implementation of ```ICandlesChecker```, which never notifies. And check ```NotifyOnCantStoreAssetPair``` only while the DI container configuration.